### PR TITLE
Make VScode associate .C files to cpp so we can use the plugin snippets

### PR DESCRIPTION
**Pull request by GiudGiud**
refs #21485

### CODE REVIEW

**Review by dschwen — DISMISSED** [review]

File: modules/doc/content/help/development/VSCode.md [hunk 1]
```diff
@@ -70,6 +70,9 @@ The following settings are recommended to pass the MOOSE GitHub CI pre-checks:
         "${env.CONDA_PREFIX}/**",
         "/usr/include/**"
     ],
+    "files.associations": {
+        "*.C": "cpp",
+    },
     "workbench.editorAssociations": {
         "*.C": "cpp",
         "*.h": "cpp"
```
Comment (dschwen) [inline modules/doc/content/help/development/VSCode.md:78]: I think those are conceptually wrong. The editor is not a file type, but a UI component that can be fundamentally different from a text editor. E.g. flame graph viewer for profile files.

File: modules/doc/content/help/development/VSCode.md [hunk 2]
```diff
@@ -70,6 +70,9 @@ The following settings are recommended to pass the MOOSE GitHub CI pre-checks:
         "${env.CONDA_PREFIX}/**",
         "/usr/include/**"
     ],
+    "files.associations": {
+        "*.C": "cpp",
```
Comment (dschwen) [inline modules/doc/content/help/development/VSCode.md:74]:
```suggestion
        "*.C": "cpp",
        "*.h", "cpp"
```

I have that in my config. We have snippets for headers as well. Do those work for you without this line?

Comment (GiudGiud) [reply modules/doc/content/help/development/VSCode.md:74]: yeah they do

Comment (GiudGiud) [reply modules/doc/content/help/development/VSCode.md:74]: adding this removes a zillion other suggestions from the list, which all look useless in our context (bunch of ifdef for all types of things)

let s do it?>

File: modules/doc/content/help/development/VSCode.md (hunk 1 above)
Comment (GiudGiud) [reply modules/doc/content/help/development/VSCode.md:78]: well they rely on it so I dont think we ought to change this here

**Review by dschwen — COMMENTED** [review]
see comment

**Comment by GiudGiud** [comment]
see which comment I replied

File: modules/doc/content/help/development/VSCode.md (hunk 1 above)
Comment (dschwen) [reply modules/doc/content/help/development/VSCode.md:78]: Who relies on it? These settings are invalid:

image

```
Value is not accepted. Valid values: "default", "workbench.editorinputs.searchEditorInput", "workbench.editor.chatSession", "jupyter-notebook", "vscode.markdown.preview.editor", "imagePreview.previewEditor", "vscode.audioPreview", "vscode.videoPreview", "jsProfileVisualizer.cpuprofile.table", "jsProfileVisualizer.heapprofile.table", "amiga.profile", "amiga.objdump", "amiga.savestate", "gitlens.rebase", "workbench.editors.gettingStartedInput".(1)
```

File: modules/doc/content/help/development/VSCode.md (hunk 2 above)
Comment (dschwen) [reply modules/doc/content/help/development/VSCode.md:74]: I'll leave that up to you. If snippets work that's fine. I have the setting  in my config anyways.

File: modules/doc/content/help/development/VSCode.md (hunk 1 above)
Comment (GiudGiud) [reply modules/doc/content/help/development/VSCode.md:78]: alright I ll delete them

File: modules/doc/content/help/development/VSCode.md (hunk 2 above)
Comment (GiudGiud) [reply modules/doc/content/help/development/VSCode.md:74]: updated

**Review by dschwen — APPROVED** [review]

**Pull request merged**